### PR TITLE
[#157929923] add-badge-messages

### PR DESCRIPTION
**Pull request by matgentili**
[#157929923] not completed. For the moment the badge is visible in all the tabs. Logic for badge is not implemented

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#157929923](https://www.pivotaltracker.com/story/show/157929923): Implementare badge nel tab dei messaggi con il numero di messaggi non letti


New dependencies added: <a href='https://www.npmjs.com/package/@types/react-native-elements'><code>&#64;types/react-native-elements</code></a>, <a href='https://www.npmjs.com/package/tslint-config-prettier'><code>tslint-config-prettier</code></a>, <a href='https://www.npmjs.com/package/tslint-eslint-rules'><code>tslint-eslint-rules</code></a>, <a href='https://www.npmjs.com/package/tslint-immutable'><code>tslint-immutable</code></a>, <a href='https://www.npmjs.com/package/tslint-plugin-prettier'><code>tslint-plugin-prettier</code></a>, <a href='https://www.npmjs.com/package/tslint-react'><code>tslint-react</code></a>, <a href='https://www.npmjs.com/package/tslint-sonarts'><code>tslint-sonarts</code></a> and <a href='https://www.npmjs.com/package/typestrict'><code>typestrict</code></a>.


<h2><a href="http://npmjs.com/package/@types/react-native-elements">&#64;types/react-native-elements</a></h2>
<p>Author: Unknown</p>
<p>Description: Stub TypeScript definitions entry for react-native-elements, which provides its own types definitions</p>
<p>Homepage: <a href="http://npmjs.com/package/@types/react-native-elements">http://npmjs.com/package/@types/react-native-elements</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 2 years ago</td></tr><tr><td>Last Updated</td><td>over 1 year ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>9</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/react-native-elements'><code>react-native-elements</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

This is a stub types definition for react-native-elements (https://github.com/react-native-training/react-native-elements).
react-native-elements provides its own type definitions, so you don't need @types/react-native-elements installed!

</details>




<h2><a href="https://github.com/prettier/tslint-config-prettier#readme">tslint-config-prettier</a></h2>
<p>Author: Alex Jover Morales</p>
<p>Description: Do you wanna use tslint and prettier without conflicts? tslint-config-prettier disables all conflicting rules that may cause such problems. Prettier takes care of formatting and tslint the rest.</p>
<p>Homepage: <a href="https://github.com/prettier/tslint-config-prettier#readme">https://github.com/prettier/tslint-config-prettier#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 2 years ago</td></tr><tr><td>Last Updated</td><td>3 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>19</td></tr><tr><td>Keywords</td><td>lint, tslint, ts-lint, prettier, config and typescript</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# tslint-config-prettier

[npm](https://www.npmjs.com/package/tslint-config-prettier)
[Travis](https://travis-ci.org/prettier/tslint-config-prettier)
[downloads](https://www.npmjs.com/package/tslint-config-prettier)
[dependencies Status](https://david-dm.org/prettier/tslint-config-prettier)
[devDependencies Status](https://david-dm.org/prettier/tslint-config-prettier?type=dev)

<h3> :cop: TSLint  +  :nail_care: Prettier = :heart_eyes: </h3>

Do you want to use [TSLint](https://palantir.github.io/tslint/) and [Prettier](https://github.com/prettier/prettier) without conflicts? `tslint-config-prettier` disables all [conflicting rules](https://unpkg.com/tslint-config-prettier) that may cause such problems. Prettier takes care of the formatting whereas tslint takes care of all the other things.

> Check how it works in **[this tutorial](https://alexjoverm.github.io/2017/06/12/Use-Prettier-with-TSLint-and-be-happy/)**.

### Get started

```bash
yarn add --dev tslint-config-prettier
# or
npm install --save-dev tslint-config-prettier
```

Make sure you've already set up [TSLint](https://palantir.github.io/tslint/) and [Prettier](https://github.com/prettier/prettier).

Then, extend your `tslint.json`, and make sure `tslint-config-prettier` is **at the end**:

```json
{
  "extends": [
    "tslint:latest",
    "tslint-config-prettier"
  ]
}
```

### More configuration

`tslint-config-prettier` also turns off formatting rules from the following rulesets, so you can use them safely.

- [codelyzer](https://github.com/mgechev/codelyzer)
- [tslint](https://github.com/palantir/tslint)
- [tslint-consistent-codestyle](https://github.com/ajafff/tslint-consistent-codestyle)
- [tslint-divid](https://github.com/jonaskello/tslint-divid)
- [tslint-eslint-rules](https://github.com/buzinas/tslint-eslint-rules)
- [tslint-immutable](https://github.com/jonaskello/tslint-immutable)
- [tslint-microsoft-contrib](https://github.com/Microsoft/tslint-microsoft-contrib)
- [tslint-misc-rules](https://github.com/jwbay/tslint-misc-rules)
- [tslint-react](https://github.com/palantir/tslint-react)
- [vrsource-tslint-rules](https://github.com/vrsource/vrsource-tslint-rules)

```json
{
  "extends": [
    "tslint:latest",
    "tslint-react",
    "tslint-eslint-rules",
    "tslint-config-prettier"
  ]
}
```

### CLI helper tool

`tslint-config-prettier` is shipped with a little CLI tool to help you check if your configuration contains any rules that are in conflict with Prettier. (require `tslint` installed)

In order to execute the CLI tool, first add a script for it to `package.json`:

```json
{
  "scripts": {
    "tslint-check": "tslint-config-prettier-check ./tslint.json"
  }
}
```

Then run `yarn tslint-check` or `npm run tslint-check`.

### Tutorials

- [Using TSlint with Prettier](https://alexjoverm.github.io/2017/06/12/Use-Prettier-with-TSLint-and-be-happy/)
- [Use Prettier with TSLint without conflicts (video)](https://egghead.io/lessons/typescript-use-prettier-with-tslint-without-conflicts-c39670eb/)

### Contributing

Please read [CONTRIBUTING.md](https://github.com/prettier/tslint-config-prettier/blob/master/CONTRIBUTING.md) for details on our code of conduct, and the process for submitting pull requests to us.

### Related

- [tslint-plugin-prettier](https://github.com/ikatyang/tslint-plugin-prettier) - Runs Prettier as a TSLint rule and reports differences as individual TSLint issues.

### Credits

Made with :heart: by [@alexjoverm](https://twitter.com/alexjoverm) and all its [contributors](https://github.com/prettier/tslint-config-prettier/graphs/contributors)


</details>




<h2><a href="http://npmjs.com/package/tslint-eslint-rules">tslint-eslint-rules</a></h2>
<p>Author: Vitor Buzinaro</p>
<p>Description: Improve your TSLint with the missing ESLint Rules</p>
<p>Homepage: <a href="http://npmjs.com/package/tslint-eslint-rules">http://npmjs.com/package/tslint-eslint-rules</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 3 years ago</td></tr><tr><td>Last Updated</td><td>8 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>2</td></tr><tr><td>Releases</td><td>45</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/doctrine'><code>doctrine</code></a>, <a href='https://www.npmjs.com/package/tslib'><code>tslib</code></a> and <a href='https://www.npmjs.com/package/tsutils'><code>tsutils</code></a></td></tr><tr><td>Keywords</td><td>eslint, tslint, custom-rules, rules, stylish, lint, linting, linter and tslint-plugin</td></tr>
</table>
This README is too long to show.



<h2><a href="https://github.com/jonaskello/tslint-immutable#readme">tslint-immutable</a></h2>
<p>Author: Jonas Kello</p>
<p>Description: TSLint rules to disable mutation in TypeScript.</p>
<p>Homepage: <a href="https://github.com/jonaskello/tslint-immutable#readme">https://github.com/jonaskello/tslint-immutable#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>about 3 years ago</td></tr><tr><td>Last Updated</td><td>about 1 month ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>2</td></tr><tr><td>Releases</td><td>54</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/tsutils'><code>tsutils</code></a></td></tr><tr><td>Keywords</td><td>tslint and immutability</td></tr>
</table>
This README is too long to show.



<h2><a href="https://github.com/ikatyang/tslint-plugin-prettier#readme">tslint-plugin-prettier</a></h2>
<p>Author: Ika</p>
<p>Description: Runs Prettier as a TSLint rule and reports differences as individual TSLint issues.</p>
<p>Homepage: <a href="https://github.com/ikatyang/tslint-plugin-prettier#readme">https://github.com/ikatyang/tslint-plugin-prettier#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 1 year ago</td></tr><tr><td>Last Updated</td><td>6 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>6</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/eslint-plugin-prettier'><code>eslint-plugin-prettier</code></a>, <a href='https://www.npmjs.com/package/lines-and-columns'><code>lines-and-columns</code></a> and <a href='https://www.npmjs.com/package/tslib'><code>tslib</code></a></td></tr><tr><td>Keywords</td><td>prettier, tslint, tslint-plugin and tslint-rules</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# tslint-plugin-prettier

[npm](https://www.npmjs.com/package/tslint-plugin-prettier)
[build](https://travis-ci.org/ikatyang/tslint-plugin-prettier/builds)
[coverage](https://codecov.io/gh/ikatyang/tslint-plugin-prettier)
[dependencies](https://david-dm.org/ikatyang/tslint-plugin-prettier)
[devDependencies](https://david-dm.org/ikatyang/tslint-plugin-prettier?type=dev)

Runs Prettier as a TSLint rule and reports differences as individual TSLint issues.

[Changelog](https://github.com/ikatyang/tslint-plugin-prettier/blob/master/CHANGELOG.md)

## Sample

```ts
a();;;
    ~~
;;;
~~~ [Delete `;;⏎;;;`]
```

```ts
var foo = ''
          ~~ [Replace `''` with `"";⏎`]
```

```ts
var foo= "";
       ~ [Insert `·`]
```

## Install

```sh
# using npm
npm install --save-dev tslint-plugin-prettier prettier

# using yarn
yarn add --dev tslint-plugin-prettier prettier
```

(require `prettier@^1.9.0`)

## Usage

(tslint.json)

for `tslint@^5.0.0`

```json
{
  "extends": ["tslint-plugin-prettier"],
  "rules": {
    "prettier": true
  }
}
```

for `tslint@^5.2.0`

```json
{
  "rulesDirectory": ["tslint-plugin-prettier"],
  "rules": {
    "prettier": true
  }
}
```

**NOTE**: To use this plugin, it'd better to also use [tslint-config-prettier](https://github.com/alexjoverm/tslint-config-prettier) to disable all prettier-related rules, so as to avoid conflicts between existed rules.

## Options

- If there is no option provided, it'll try to load [config file](https://prettier.io/docs/en/configuration.html) and/or `.editorconfig` if possible, uses Prettier's default option if not found.

  ```json
  {
    "extends": ["tslint-plugin-prettier"],
    "rules": {
      "prettier": true
    }
  }
  ```

  If you don't want to load `.editorconfig`, disable it in the third argument.

  ```json
  {
    "extends": ["tslint-plugin-prettier"],
    "rules": {
      "prettier": [true, null, { "editorconfig": false }]
    }
  }
  ```

- If you'd like to specify which config file to use, just put its path (relative to `process.cwd()`) in the second argument, the following example shows how to load the config file from `<cwd>/configs/.prettierrc`:

  ```json
  {
    "extends": ["tslint-plugin-prettier"],
    "rules": {
      "prettier": [true, "configs/.prettierrc"]
    }
  }
  ```

- If you'd like to specify options manually, just put [Prettier Options](https://prettier.io/docs/en/options.html) in the second argument, for example:

  ```json
  {
    "extends": ["tslint-plugin-prettier"],
    "rules": {
      "prettier": [true, { "singleQuote": true }]
    }
  }
  ```

## Development

```sh
# lint
yarn run lint

# build
yarn run build

# test
yarn run test
```

## Related

- [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier)
- [tslint-config-prettier](https://github.com/alexjoverm/tslint-config-prettier)
- [prettier-tslint](https://github.com/azz/prettier-tslint)

## License

MIT © [Ika](https://github.com/ikatyang)


</details>




<h2><a href="https://github.com/palantir/tslint-react#readme">tslint-react</a></h2>
<p>Author: Unknown</p>
<p>Description: Lint rules related to React & JSX for TSLint</p>
<p>Homepage: <a href="https://github.com/palantir/tslint-react#readme">https://github.com/palantir/tslint-react#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 3 years ago</td></tr><tr><td>Last Updated</td><td>28 days ago</td></tr><tr><td>License</td><td>Apache-2.0</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>26</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/tsutils'><code>tsutils</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

[NPM version](https://www.npmjs.com/package/tslint-react)
[Downloads](https://npmjs.org/package/tslint-react)
[Circle CI](https://circleci.com/gh/palantir/tslint-react)

tslint-react
------------

Lint rules related to React & JSX for [TSLint](https://github.com/palantir/tslint/).

### Usage

tslint-react has peer dependencies on TSLint and TypeScript.

To use these lint rules with the default preset, use configuration inheritance via the `extends` keyword.
Here's a sample configuration where `tslint.json` lives adjacent to your `node_modules` folder:

```js
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    // override tslint-react rules here
    "jsx-wrap-multiline": false
  }
}
```

To lint your `.ts` **and** `.tsx` files you can simply run `tslint -c tslint.json 'src/**/*.{ts,tsx}'`.

### Semantic versioning

The built-in configuration preset you get with `"extends": "tslint-react"` is semantically versioned in a manner similar to TSLint's built-in presets and the TypeScript language itself. As new rules are added to tslint-react across minor versions, stricter checks may be enabled here. Your code is not guaranteed to continue passing checks across these version bumps. If you wish to ensure that `npm upgrade` or `yarn upgrade` never breaks your build, declare a tilde dependency on this package (e.g. `"~1.0.0"`).

### Rules

- `jsx-alignment`
  - Enforces a consistent style for multiline JSX elements which promotes ease of editing via line-wise manipulations
  as well as maintainability via small diffs when changes are made.
  ```ts
  // Good:
  const element = <div
      className="foo"
      tabIndex={1}
  >
      {children}
  </div>;

  // Also Good:
  <Button
      appearance="pretty"
      disabled
      label="Click Me"
      size={size}
  />
  ```
- `jsx-ban-elements` (since v3.4.0)
  - Allows blacklisting of JSX elements with an optional explanatory message in the reported failure.
- `jsx-ban-props` (since v2.3.0)
  - Allows blacklisting of props in JSX with an optional explanatory message in the reported failure.
- `jsx-boolean-value` (since v2.5.0)
  - When using a boolean attribute in JSX, you can set the attribute value to true or omit the value. This rule will enforce one or the other to keep consistency in your code.
  - Rule options: `["always", "never"]`
  - Default is set to `always`.
- `jsx-curly-spacing` (since v1.1.0)
  - Requires _or_ bans spaces between curly brace characters in JSX.
  - Rule options: `["always", "never"]`
  - _Includes automatic code fix_
- `jsx-equals-spacing` (since v3.2.0)
  - Requires _or_ bans spaces before and after the `=` token in JSX element attributes.
  - Rule options: `["always", "never"]`
  - _Includes automatic code fix_
- `jsx-key` (since v3.2.0)
  - Warns for missing `key` props in JSX element array literals and inside return statements of `Array.prototype.map` callbacks.
    - N.B. This rule only does a simple check for `.map(...)` syntax and does not inspect computed types of expressions. As such, it may produce false positives if you use APIs that look similar to `.map()`.
  - Rule options: _none_
- `jsx-no-bind` (since v2.6.0)
  - Forbids function binding in JSX attributes. This has the same intent as `jsx-no-lambda` in helping you avoid excessive re-renders.
  - Note that this currently only does a simple _syntactic_ check, not a _semantic_ one (it doesn't use the type checker). So it may have some
    rare false positives if you define your own `.bind` function and supply `this` as a parameter.
  - Rule options: _none_
- `jsx-no-lambda`
  - Creating new anonymous functions (with either the `function` syntax or ES2015 arrow syntax) inside the `render` call stack works against _pure component rendering_. When doing an equality check between two lambdas, React will always consider them unequal values and force the component to re-render more often than necessary.
  - Rule options: _none_
- `jsx-no-multiline-js`
  - Disallows multiline JS expressions inside JSX blocks to promote readability
  - Rule options: _none_
- `jsx-no-string-ref`
  - Passing strings to the `ref` prop of React elements is considered a legacy feature and will soon be deprecated.
    Instead, [use a callback](https://facebook.github.io/react/docs/more-about-refs.html#the-ref-callback-attribute).
  - Rule options: _none_
- `jsx-use-translation-function` (since v2.4.0)
   - Enforces use of a translation function. Plain string literals are disallowed in JSX when enabled.
   - Rule options: `["allow-punctuation", "allow-htmlentities"]`
   - Off by default
- `jsx-self-close` (since v0.4.0)
  - Enforces that JSX elements with no children are self-closing.
  - _Includes automatic code fix_
  ```ts
  // bad
  <div className="foo"></div>
  // good
  <div className="foo" />
  ```
  - Rule options: _none_
- `jsx-space-before-trailing-slash`
  - Checks that self-closing JSX elements have a space before the '/>' part.
  - Rule options: _none_
  - _Includes automatic code fix_
- `jsx-wrap-multiline` (since v2.1)
  - Enforces that multiline JSX expressions are wrapped with parentheses.
  - Opening parenthesis must be followed by a newline.
  - Closing parenthesis must be preceded by a newline.
  ```ts
  // bad
  const button = <button type="submit">
      Submit
  </button>;
  // good
  const button = (
      <button type="submit">
          Submit
      </button>
  );
  ```

### Development

We track rule suggestions on Github issues -- [here's a useful link](https://github.com/palantir/tslint-react/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Rule+Suggestion%22)
to view all the current suggestions. Tickets are roughly triaged by priority (P1, P2, P3).

We're happy to accept PRs for new rules, especially those marked as [Status: Accepting PRs](https://github.com/palantir/tslint-react/issues?q=is%3Aissue+is%3Aopen+label%3A%22Status%3A+Accepting+PRs%22).
If submitting a PR, try to follow the same style conventions as the [core TSLint project](https://github.com/palantir/tslint).

Quick Start (requires Node v6+, yarn v0.22+):

1. `yarn`
1. `yarn verify`
1. `yarn lint`

### Changelog

See the Github [release history](https://github.com/palantir/tslint-react/releases).


</details>




<h2><a href="https://github.com/SonarSource/SonarTS">tslint-sonarts</a></h2>
<p>Author: Unknown</p>
<p>Description: SonarTS rules for TSLint</p>
<p>Homepage: <a href="https://github.com/SonarSource/SonarTS">https://github.com/SonarSource/SonarTS</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 2 years ago</td></tr><tr><td>Last Updated</td><td>3 months ago</td></tr><tr><td>License</td><td>LGPL-3.0</td></tr><tr><td>Maintainers</td><td>4</td></tr><tr><td>Releases</td><td>37</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/immutable'><code>immutable</code></a></td></tr><tr><td>Keywords</td><td>sonarts, sonarqube, typescript and tslint</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

ERROR: No README data found!

</details>




<h2><a href="https://github.com/krzkaczor/TypeStrict#readme">typestrict</a></h2>
<p>Author: Unknown</p>
<p>Description: TSLint config focused on maximizing type safety</p>
<p>Homepage: <a href="https://github.com/krzkaczor/TypeStrict#readme">https://github.com/krzkaczor/TypeStrict#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 1 year ago</td></tr><tr><td>Last Updated</td><td>4 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>11</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/tslint-microsoft-contrib'><code>tslint-microsoft-contrib</code></a> and <a href='https://www.npmjs.com/package/tslint-sonarts'><code>tslint-sonarts</code></a></td></tr><tr><td>Keywords</td><td>typescript, tslint, type safety, ruleset and types</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

<p align="center">
  TypeStrict
  <h3 align="center">TypeStrict</h3> 
  <p align="center">Squash bugs before shipping them with TypeScript and TSLint!</p>
  <p align="center">TSLint config focused on maximizing type safety</p>
  <p align="center">
    Downloads
    <a href="/package.json">Software License</a>
  </p>

</p>

# Installation

```bash
npm install --save-dev typestrict
```

or

```bash
yarn add --dev typestrict
```

# Usage

Create `tslint.json`:

```json
{
  "extends": "typestrict"
  // ... add here your own rules
}
```

and run `tslint -c ./tslint.json --project ./tsconfig.json --format stylish`.

Remember to use `strict` compiler setting in `tsconfig.json`!

_Please, for now ignore warning about deprecated rule, it's discussed [here](https://github.com/krzkaczor/TypeStrict/pull/4)_

# Description

All rules in this config are focused on catching bugs in your code and be as not annoying as possible. They don't
care about code style or formatting.

`TypeStrict` combines rules from various rule sets: [tslint](https://palantir.github.io/tslint/rules/),
[sonarTS](https://github.com/SonarSource/SonarTS),
[tslint-microsoft-contrib](https://github.com/Microsoft/tslint-microsoft-contrib).

## Complete rules list

### TSLint

- [await-promise](https://palantir.github.io/tslint/rules/await-promise/)
- [no-floating-promises](https://palantir.github.io/tslint/rules/no-floating-promises/)
- [no-unused-variable](https://palantir.github.io/tslint/rules/no-unused-variable/) (with `"check-parameters"` and `{"ignore-pattern": "^_" }`)
- [no-use-before-declare](https://palantir.github.io/tslint/rules/no-use-before-declare/)
- [no-duplicate-super](https://palantir.github.io/tslint/rules/no-duplicate-super/)
- [no-inferred-empty-object-type](https://palantir.github.io/tslint/rules/no-inferred-empty-object-type/)
- [no-invalid-this](https://palantir.github.io/tslint/rules/no-invalid-this/)
- [no-misused-new](https://palantir.github.io/tslint/rules/no-misused-new/)
- [restrict-plus-operands](https://palantir.github.io/tslint/rules/restrict-plus-operands/)
- [use-isnan](https://palantir.github.io/tslint/rules/use-isnan/)

### SonarTS

- [no-accessor-field-mismatch](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-accessor-field-mismatch.md)
- [no-all-duplicated-branches](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-all-duplicated-branches.md)
- [no-collection-size-mischeck](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-collection-size-mischeck.md)
- [no-dead-store](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-dead-store.md)
- [no-duplicated-branches](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-duplicated-branches.md)
- [no-element-overwrite](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-element-overwrite.md)
- [no-empty-destructuring](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-empty-destructuring.md)
- [no-gratuitous-expressions](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-gratuitous-expressions.md)
- [no-identical-conditions](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-identical-conditions.md)
- [no-identical-expressions](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-identical-expressions.md)
- [no-ignored-return](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-ignored-return.md)
- [no-multiline-string-literals](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-multiline-string-literals.md)
- [no-self-assignment](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-self-assignment.md)
- [no-unconditional-jump](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-unconditional-jump.md)
- [no-unthrown-error](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-unthrown-error.md)
- [no-unused-array](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-unused-array.md)
- [no-useless-cast](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-useless-cast.md)
- [no-useless-increment](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-useless-increment.md)
- [no-variable-usage-before-declaration](https://github.com/SonarSource/SonarTS/blob/HEAD/sonarts-core/docs/rules/no-variable-usage-before-declaration.md)

### Microsoft Contrib

- [promise-must-complete](https://github.com/Microsoft/tslint-microsoft-contrib/issues/34)

# Contributing

Do you have idea for new rule to be added? You don't agree with one of rules included? Feel free to create issue / PR.

# License

MIT by [Krzysztof Kaczor](https://twitter.com/krzKaczor)


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by fpersico** [comment]
@matgentili @cloudify  We are using native-base as UI library and it also has a Badge component. I don't think we need to add another UI library like react-native-elements.

**Comment by matgentili** [comment]
@fpersico ok, i remove react-native-elements library and use native-base.

File: ts/components/messages/MessageListComponent.tsx [hunk 1]
```diff
@@ -71,7 +71,6 @@ class MessageListComponent extends React.Component<Props> {
     const refreshControl = (
       <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
     );
-
```
Comment (cloudify) [inline ts/components/messages/MessageListComponent.tsx]: why this change?

File: ts/navigation/MainNavigator.tsx [hunk 2]
```diff
@@ -1,12 +1,6 @@
-/**
```
Comment (cloudify) [inline ts/navigation/MainNavigator.tsx]: why you removed the comment?

File: ts/components/messages/MessageListComponent.tsx (hunk 1 above)
Comment (matgentili) [reply ts/components/messages/MessageListComponent.tsx]: Sorry, i think i removed the line during implementing an other const that i removed later

File: ts/navigation/MainNavigator.tsx (hunk 2 above)
Comment (matgentili) [reply ts/navigation/MainNavigator.tsx]: Sorry, I didn't even notice it. Most likely I removed it while importing a library

File: package.json [hunk 3]
```diff
@@ -99,6 +99,7 @@
     "@types/react": "16.8.10",
     "@types/react-native": "0.57.42",
     "@types/react-native-background-timer": "^2.0.0",
+    "@types/react-native-elements": "^0.18.0",
```
Comment (cloudify) [inline package.json]: this need to be removed too

File: ts/components/messages/MessageListComponent.tsx [hunk 4]
```diff
@@ -12,7 +12,8 @@ import { MessageState } from "../../store/reducers/entities/messages/messagesByI
 import { PaymentByRptIdState } from "../../store/reducers/entities/payments";
 import { ServicesByIdState } from "../../store/reducers/entities/services/servicesById";
 import { MessageListItemComponent } from "./MessageListItemComponent";
-
+// tslint:disable-next-line: no-var-keyword
+var messagesToRead = 0;
```
Comment (cloudify) [inline ts/components/messages/MessageListComponent.tsx]: don't use vars, the tslint rule is there for a reason :)

File: ts/components/messages/MessageListComponent.tsx [hunk 5]
```diff
@@ -67,11 +68,13 @@ class MessageListComponent extends React.Component<Props> {
       paymentByRptId,
       ListEmptyComponent
     } = this.props;
-
     const refreshControl = (
       <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
     );
 
+    messagesToRead = messages.filter(obj => !obj.isRead).length;
+    alert(messagesToRead);
```
Comment (cloudify) [inline ts/components/messages/MessageListComponent.tsx]: what's this alert?

File: yarn.lock [hunk 6]
```diff
@@ -1071,6 +1071,13 @@
   version "2.0.0"
```
Comment (cloudify) [inline yarn.lock]: since you didn't add any package, `yarn.lock` must not change

File: package.json (hunk 3 above)
Comment (matgentili) [reply package.json]: Ok

File: ts/components/messages/MessageListComponent.tsx (hunk 4 above)
Comment (matgentili) [reply ts/components/messages/MessageListComponent.tsx]: Ok, but if i don't use vars what kind of variable should I use to update badge?

File: yarn.lock (hunk 6 above)
Comment (matgentili) [reply yarn.lock]: Ok!

File: ts/components/messages/MessageListComponent.tsx (hunk 5 above)
Comment (matgentili) [reply ts/components/messages/MessageListComponent.tsx]: Sorry it was only for tests and I forgot to remove it

File: ts/navigation/MainNavigator.tsx [hunk 7]
```diff
@@ -178,20 +180,27 @@ const navigation = createBottomTabNavigator(
       tabBarIcon: (options: { tintColor: string | null; focused: boolean }) => {
         const { routeName } = nav.state;
         const iconName: string = getIcon(routeName);
+        const messagesToRead = MessageListComponent.prototype.getMessagesToRead();
```
Comment (cloudify) [inline ts/navigation/MainNavigator.tsx]: can you explain how this works?

Comment (matgentili) [reply ts/navigation/MainNavigator.tsx]: I take from the component that manages the list of messages the number of unread messages and save them in a constant which is then used to update the view. Do you think that could be fine?

Comment (cloudify) [reply ts/navigation/MainNavigator.tsx]: I don't think it could possibly work, did you tested it?

Comment (matgentili) [reply ts/navigation/MainNavigator.tsx]: Yes, I have tested it on ios simulator and on android device and when I open a message the badge number is updated correctly.

**Comment by cloudify** [comment]
@matgentili you should create a custom component that subscribes to the redux store to get the list of messages

**Comment by cloudify** [comment]
@matgentili any update here?

**Comment by matgentili** [comment]
@cloudify I used Redux store as you suggested, now i'm testing it and soon I commit all ;)

**Comment by matgentili** [comment]
badge_message

File: ts/components/MessagesTabIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    bottom: 10
+  }
+});
+
+type Props = OwnProps & ReturnType<typeof mapStateToProps>;
+
+/**
+ * Filters the list of messages and returns the number of unread messages.
+ */
+const getNumberMessagesUnread = (
```
Comment (fpersico) [inline ts/components/MessagesTabIcon.tsx]: Convert this to a redux selector

File: ts/components/MessagesTabIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+      <View>
+        <IconFont
+          name={"io-messaggi"}
+          size={variables.iconSize3}
+          color={color}
+        />
+        {messagesUnread > 0 ? (
+          Platform.OS === "ios" ? (
+            <Badge style={styles.badgeStyle}>
+              <Text style={[styles.textBadgeStyle, { top: 0 }]}>
```
Comment (fpersico) [inline ts/components/MessagesTabIcon.tsx]: No inline style

File: ts/components/MessagesTabIcon.tsx (hunk 8 above)
Comment (fpersico) [inline ts/components/MessagesTabIcon.tsx]: Convert this to a redux selector

File: ts/components/MessagesTabIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+          )
+        ) : null}
+      </View>
+    );
+  }
+}
+
+const mapStateToProps = (state: GlobalState) => ({
+  messagesUnread:
+    getNumberMessagesUnread(lexicallyOrderedMessagesStateSelector(state)) < 99
```
Comment (fpersico) [inline ts/components/MessagesTabIcon.tsx]: You are calling the function twice!

File: ts/navigation/MainNavigator.tsx (hunk 2 above)
Comment (fpersico) [reply ts/navigation/MainNavigator.tsx]: Still removed

File: ts/navigation/MainNavigator.tsx [hunk 11]
```diff
@@ -87,6 +79,30 @@ const styles = StyleSheet.create({
     shadowRadius: variables.footerShadowRadius,
     // Android shadow
     elevation: variables.footerElevation
+  },
+  textBadgeStyle: {
```
Comment (fpersico) [inline ts/navigation/MainNavigator.tsx]: Not used anymore?

**Comment by fpersico** [comment]
@matgentili Please clean this PR because it contains unused stuff.

File: yarn.lock (hunk 6 above)
Comment (fpersico) [reply yarn.lock]: @matgentili please check this.

Comment (matgentili) [reply yarn.lock]: @fpersico Can i remove yarn.lock ? I don't add any package

Comment (fpersico) [reply yarn.lock]: Just revert the changes.
